packages/connect-wallet-modal/src/components/WalletsModalForEth/WalletsModalForEth.tsx

```diff
@@ -106,9 +108,11 @@ function getWalletsButtons(
   // TODO: add better wallets ordering when all wallets migrated to wallet adapter API
   const okxWalletId = 'okx';
   const okxWalletIndex = wallets.indexOf(okxWalletId);
-  if (okxWalletIndex >= 0) {
-    wallets.splice(okxWalletIndex, 1);
-    wallets.splice(1, 0, okxWalletId);
+  const metamaskIndex = wallets.indexOf(WALLET_IDS.METAMASK);
+  // If metamask is there and okx was initially placed more than one step away from metamask
+  if (metamaskIndex > -1 && okxWalletIndex > metamaskIndex + 1) {
+    wallets.splice(okxWalletIndex, 1); // remove okx from its place
+    wallets.splice(metamaskIndex + 1, 0, okxWalletId); // place okx right after metamask
   }
```

Comment:
please figure out later how to preserve this logic in OKX or Metamask adapter, not in general code